ccdproc/image_collection.py

```diff
@@ -0,0 +1,643 @@
+            if value == '*':
+                have_this_value = value_not_missing
+            elif value is not None:
+                if isinstance(value, six.string_types):
+                    # need to loop explicitly over array rather than using
+                    # where to correctly do string comparison.
+                    have_this_value = np.array([False] * len(use_info))
+                    for idx, file_key_value in enumerate(use_info[key]):
+                        if value_not_missing[idx]:
+                            value_matches = (file_key_value.lower() ==
+                                             value.lower())
+                        else:
+                            value_matches = False
+
+                        have_this_value[idx] = (value_not_missing[idx] &
+                                                value_matches)
+                else:
+                    have_this_value = value_not_missing
+                    tmp = (use_info[key][value_not_missing] == value)
+                    have_this_value[value_not_missing] = tmp
+                    have_this_value &= value_not_missing
+            else:
+                # this case--when value==None--is asking for the files which
+                # are missing a value for this keyword
+                have_this_value = value_missing
+
+            matches &= have_this_value
+
+        # the numpy convention is that the mask is True for values to
+        # be omitted, hence use ~matches.
+        logger.debug('Matches: %s', matches)
+        self.summary_info['file'].mask = ma.nomask
+        self.summary_info['file'][~matches] = ma.masked
+
+    def _fits_files_in_directory(self, extensions=None,
+                                 compressed=True):
+        """
+        Get names of FITS files in directory, based on filename extension.
+
+        Parameters
+        ----------
+        extension : list of str, optional
+            List of filename extensions that are FITS files. Default is
+            ``['fit', 'fits']``
+
+        compressed : bool, optional
+            If ``True``, compressed files should be included in the list
+            (e.g. `.fits.gz`)
+
+        Returns
+        -------
+        list
+            *Names* of the files (with extension), not the full pathname.
+        """
+        full_extensions = extensions or ['fit', 'fits']
+        if compressed:
+            with_gz = [extension + '.gz' for extension in full_extensions]
+            full_extensions.extend(with_gz)
+
+        all_files = listdir(self.location)
+        files = []
+        for extension in full_extensions:
+            files.extend(fnmatch.filter(all_files, '*' + extension))
+
+        return files
+
+    def _generator(self, return_type,
+                   save_with_name="", save_location='',
+                   clobber=False,
+                   overwrite=False,
+                   do_not_scale_image_data=True,
```

Comment:
Actually, `clobber` is only there for backwards compatibility. I'd prefer to move to overwrite, which seems clearer to the uninitiated.